src/os/filestore/FileStore.cc

```diff
@@ -2101,6 +2102,16 @@ int FileStore::queue_transactions(Sequencer *posr, vector<Transaction>& tls,
     dout(5) << "queue_transactions new " << osr << " " << *osr << dendl;
   }
 
+  bool store_locked = false;
+  if (cct->_conf->osd_enable_pglock_fast_release && osd_op.get()
+        && osd_op->get_pglock_can_release_flag()) {
+      // obtain the store lock
+      osd_op->get_pg()->lockstore();
```

Comment:
This bit where the filestore calls back into PG by name is shattering all our interface boundaries, which is really not appropriate. Why not have the OSD code responsible for the lock twiddling?